OrbitGl/App.cpp

```diff
@@ -271,9 +269,32 @@ void OrbitApp::PostInit() {
     process_manager_ =
         ProcessManager::Create(grpc_channel_, absl::Milliseconds(1000));
 
-    auto callback = [&](ProcessManager* process_manager) {
-      main_thread_executor_->Schedule([&, process_manager]() {
-        m_ProcessesDataView->SetProcessList(process_manager->GetProcessList());
+    auto callback = [this](ProcessManager* process_manager) {
+      main_thread_executor_->Schedule([this, process_manager]() {
+        const std::vector<ProcessInfo>& process_infos =
+            process_manager->process_list();
+        data_manager_->UpdateProcessInfos(process_infos);
+        m_ProcessesDataView->SetProcessList(process_infos);
+
+        {
+          // TODO: remove this part when client stops using Process class
+          absl::MutexLock lock(&process_map_mutex_);
+          for (const ProcessInfo& info : process_infos) {
+            auto it = process_map_.find(info.pid());
+            if (it != process_map_.end()) {
+              continue;
+            }
+
+            std::shared_ptr<Process> process = std::make_shared<Process>();
+            process->SetID(info.pid());
+            process->SetName(info.name());
+            process->SetIsRemote(true);
+            // The other fields do not appear to be used at the moment.
+
+            process_map_.insert_or_assign(process->GetID(), process);
```

Comment:
suggestion: `process_map_` is only used in `FindProcessByPid(pid)` which is only used in `OnProcessSelected(pid)`, which you also touch. Maybe move this whole part into `OnProcessSelected`, then `process_map_` and `FindProcessById()` are not needed anymore and could be deleted.